apps/services/user-profile/src/app/v2/dto/user-profileDto.ts

```diff
@@ -0,0 +1,56 @@
+import { IsBoolean, IsEnum, IsOptional, IsString } from 'class-validator'
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger'
+
+import { DataStatus } from '../../user-profile/types/dataStatusTypes'
+import { DataType } from 'sequelize-typescript'
+
+export class UserProfileDto {
+  @ApiProperty()
+  @IsString()
+  readonly nationalId: string
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsString()
+  readonly email?: string
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsString()
+  readonly mobilePhoneNumber?: string
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  @IsEnum(DataType.ENUM('en', 'is'))
+  readonly locale?: string
```

Comment:
We should use the locale enum in `src/app/user-profile/types`
```suggestion
  @ApiPropertyOptional({
      enum: Locale,
      enumName: 'Locale',
  })
  @IsOptional()
  @IsEnum(Locale)
  readonly locale?: Locale
```

Reply:
I removed the profilePicture as well, as we discussed offline